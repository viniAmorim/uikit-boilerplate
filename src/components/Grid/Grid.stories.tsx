import React from 'react';

import { GridProps } from '@mantine/core';
import { Story, Meta } from '@storybook/react';

import Button from '../Button';
import Grid from './Grid';

export default {
  title: 'Atoms/Display/Grid',
  component: Grid,
} as Meta;

const Template: Story<GridProps> = (args) => <Grid {...args} />;

export const Default = Template.bind({});
Default.args = {
  bg: 'rgba(0, 0, 0, .3)',
  mih: 50,
  gutter: 50,
  children: (
    <>
      <Grid.Col span="auto" bg="blue">
        <Button variant="secondary">Button 1</Button>
      </Grid.Col>
      <Grid.Col span={6}>
        <Button variant="secondary">Button 2</Button>
      </Grid.Col>
      <Grid.Col span="auto" bg="blue">
        <Button variant="secondary">Button 3</Button>
      </Grid.Col>
    </>
  ),
};
