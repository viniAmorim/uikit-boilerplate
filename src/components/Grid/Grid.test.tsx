import React from 'react';

import { render } from '@testing-library/react';

import Grid from './Grid';

describe('Grid', () => {
  it('should render correctly', () => {
    const { container } = render(
      <Grid>
        <Grid.Col>1</Grid.Col>
        <Grid.Col>2</Grid.Col>
      </Grid>,
    );

    const grid = container.firstChild;

    expect(grid).toBeInTheDocument();
    expect(grid).toMatchSnapshot();
  });
  it('should render children with correct flexbox properties', () => {
    const { container } = render(
      <Grid justify="center" align="center">
        <Grid.Col span={4}>1</Grid.Col>
        <Grid.Col span={4}>2</Grid.Col>
        <Grid.Col span={4}>3</Grid.Col>
      </Grid>,
    );
    const flex = container.firstChild;
    expect(flex).toHaveStyle(`
      display: flex;
      justify-content: center;
      align-items: center;
    `);
  });
});
