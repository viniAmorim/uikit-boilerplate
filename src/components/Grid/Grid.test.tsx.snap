// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid should render correctly 1`] = `
<div
  class="mantine-Grid-root mantine-16fdnqw"
>
  <div
    class="mantine-Grid-col mantine-ea7fe7"
  >
    1
  </div>
  <div
    class="mantine-Grid-col mantine-ea7fe7"
  >
    2
  </div>
</div>
`;
